site-src/geps/gep-713.md

```diff
@@ -0,0 +1,496 @@
+to the following concepts on these resources:
+
+* Gateway.Listeners.Name
+* xRoute.Rules.Name
+* Service.Ports.Name
+
+```yaml
+kind: HTTPRoute
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: http-app-1
+  labels:
+    app: foo
+spec:
+  hostnames:
+  - "foo.com"
+  rules:
+  - name: bar
+    matches:
+    - path:
+        type: Prefix
+        value: /bar
+    forwardTo:
+    - serviceName: my-service1
+      port: 8080
+---
+kind: RetryPolicy
+apiVersion: networking.acme.io/v1alpha2
+metadata:
+  name: foo
+spec:
+  maxRetries: 5
+  targetRef:
+    name: foo
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    sectionName: bar
+```
+
+This would require adding a `Name` field to Gateway listeners and Route rules:
+
+```go
+type Listener struct {
+    // Name is the name of the Listener.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    // +optional
+    Name string `json:"name,omitempty"`
+    // ...
+}
+```
+
+```go
+type RouteRule struct {
+    // Name is the name of the Route rule.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    // +optional
+    Name string `json:"name,omitempty"`
+    // ...
+}
+```
+
+### Kubectl Plugin
+To help improve UX and standardization, a kubectl plugin will be developed that
+will be capable of describing the computed sum of policy that applies to a given
+resource, including policies applied to parent resources.
+
+Each Policy CRD that wants to be supported by this plugin will need to follow
+the API structure defined above and add a `gateway.networking.k8s.io/policy:
+true` label to the CRD.
+
+### Status
+In the future, we may consider adding a new `Policies` field to status on
+Gateways and Routes. This would be a list of `PolicyTargetReference` structs
+with the fields instead used to refer to the Policy resource that has been
+applied.
+
+Unfortunately, this may create more confusion than it is worth, here are some of
+the key concerns:
+
+* When multiple controllers are implementing the same Route and recognize a
+  policy, it would be difficult to determine which controller should be
+  responsible for adding that policy reference to status.
+* For this to be somewhat scalable, we'd need to limit the status entries to
+  policies that had been directly applied to the resource. This could get
+  confusing as it would not provide any insight into policies attached above or
+  below.
+* Since we only control some of the resources a policy might be attached to,
+  adding policies to status would only be possible on Gateway API resources, not
+  Services or other kinds of backends.
+
+Although these concerns are not unsolvable, they lead to the conclusion that
+a Kubectl plugin should be our primary approach to providing visibility here,
+with a possibility of adding policies to status at a later point.
```

Comment:
I agree, but I would really like to end up with some sort of status eventually. I think it makes sense to use the API for a bit before deciding on what it looks like, particularly if there's a kubectl plugin to help.

Reply:
Yep, completely agree on this one.